graphqlapi/src/main/java/io/stargate/graphql/GraphqlActivator.java

```diff
@@ -87,6 +96,19 @@ private synchronized void maybeStopService() {
       } catch (Exception e) {
         LOG.error("Unexpected error while stopping GraphQL", e);
       }
+      graphqlHealthCheck.healthy = false;
+    }
+  }
+
+  private static class GraphqlHealthCheck extends HealthCheck {
+
+    private volatile boolean healthy = false;
+
+    @Override
+    protected Result check() throws Exception {
+      return healthy
+          ? Result.healthy("Ready to process requests")
+          : Result.unhealthy("Server not started");
```

Comment:
Not sure if it's the kind of health check we'll want eventually, this is more of a proof of concept at this point.